### DISCUSSION TITLE
No significant difference in speed with or without specifying region

### DISCUSSION THREAD

**Discussion by sousvideonlow** [Spawnova/ShinsImageScanClass#6] [General]
Hey @Spawnova 

There seems to be no difference in time taken to find the image **with or without region** specified. I wonder if it was intended to be faster if you specify a region.

```
scan:= new shinsimagescanclass()
start:=A_TickCount
scan.imagearrayregion(screenshot_loc "\chart.png",objout,480,450,200,200)
end:=A_TickCount-start
Msgbox,% Obj2String(objout) "`n`n`nTime taken: " end

start:=A_TickCount
scan.image(screenshot_loc "\chart.png",,x,y)
end:=A_TickCount-start
Msgbox,% "x: " x ",y: " y "`n`n`nTime taken: " end
```

**Comment by Spawnova** [comment]
specifying a region is faster, however measuring time with a_tickcount does not generally provide very accurate results
try this instead
```AutoHotkey
DllCall("QueryPerformanceFrequency", "Int64*", freq)

scan:= new shinsimagescanclass()

DllCall("QueryPerformanceCounter", "Int64*", start)
scan.imagearrayregion(screenshot_loc "\chart.png",objout,480,450,200,200)
DllCall("QueryPerformanceCounter", "Int64*", after)
Msgbox,% Obj2String(objout) "`n`n`nTime taken: " (after-start)/freq*1000

DllCall("QueryPerformanceCounter", "Int64*", start)
scan.image(screenshot_loc "\chart.png",,x,y)
DllCall("QueryPerformanceCounter", "Int64*", after)
Msgbox,% "x: " x ",y: " y "`n`n`nTime taken: " (after-start)/freq*1000

exitapp
```
Also keep in mind `scan.ImageArray()` will be slower since it has to scan the entire screen whether an image is found or not, `scan.Image()` on the other hand stops after the first result
I have a script that runs a test on all the functions and gathers times, here's a paste to what I get on my machine - https://pastebin.com/KhJ2gDFT
The region times in the pastebin are based off scanning the entire window, in this case 2110x1126, if it was a 200x200 region instead, it would be much faster

**Reply by sousvideonlow** [reply]
Oh cool that's awesome!
